themes/default/content/blog/pulumi-targets-dotnet-6/index.md

```diff
@@ -0,0 +1,178 @@
+ - Translating Terrafrom code using [tf2pulumi](https://www.pulumi.com/tf2pulumi/)
+ - Translating Kubernetes YAML using [kube2pulumi](https://www.pulumi.com/kube2pulumi/)
+ - Translating [Pulumi YAML](https://www.pulumi.com/docs/intro/languages/yaml/) with the Pulumi CLI using `pulumi convert`
+
+All of these tools take their input language, convert it into the Pulumi intermediate language, then translate it into one of our supported languages. In case of C#, the code generation has been updated to generate code that uses latest features from .NET 6. It also fixed isuse as well as fix small issues that were present in the logic.
+
+Let us take an example from Pulumi YAML and convert it to C#.
+
+Generate a new Pulumi YAML project using `pulumi new azure-yaml`, it should give you this YAML program:
+
+```yaml
+name: azure-yaml
+runtime: yaml
+description: A minimal Azure Native Pulumi YAML program
+
+resources:
+  # Create an Azure Resource Group
+  resourceGroup:
+    type: azure-native:resources:ResourceGroup
+  # Create an Azure resource (Storage Account)
+  storageAccount:
+    type: azure-native:storage:StorageAccount
+    properties:
+      resourceGroupName: ${resourceGroup.name}
+      sku:
+        name: Standard_LRS
+      kind: StorageV2
+
+variables:
+  storageAccountKeys:
+    Fn::Invoke:
+      Function: azure-native:storage:listStorageAccountKeys
+      Arguments:
+        resourceGroupName: ${resourceGroup.name}
+        accountName: ${storageAccount.name}
+
+outputs:
+  # Export the primary key of the Storage Account
+  primaryStorageKey: ${storageAccountKeys.keys[0].value}
+```
+
+Now you can take this program and convert it to C# using
+
+```
+pulumi convert --language csharp --out csharp-from-yaml
+```
+
+This command generates a full C# project from the YAML program into the directory `./csharp-from-yaml`, then it restores the dependencies and builds the project. Now if you examine `Program.cs`, it contains the converted program and follows the succinct C# 10 syntax that targets .NET 6:
+
+```csharp
+using System.Collections.Generic;
+using Pulumi;
+using AzureNative = Pulumi.AzureNative;
+
+await Deployment.RunAsync(() => 
+{
+    var resourceGroup = new AzureNative.Resources.ResourceGroup("resourceGroup");
+
+    var storageAccount = new AzureNative.Storage.StorageAccount("storageAccount", new()
+    {
+        ResourceGroupName = resourceGroup.Name,
+        Sku = new AzureNative.Storage.Inputs.SkuArgs
+        {
+            Name = "Standard_LRS",
+        },
+        Kind = "StorageV2",
+    });
+
+    var storageAccountKeys = AzureNative.Storage.ListStorageAccountKeys.Invoke(new()
+    {
+        ResourceGroupName = resourceGroup.Name,
+        AccountName = storageAccount.Name,
+    });
+
+    return new Dictionary<string, object?>
+    {
+        ["primaryStorageKey"] = storageAccountKeys.Apply(listStorageAccountKeysResult => listStorageAccountKeysResult.Keys[0]?.Value),
+    };
+});
+```
```

Comment:
Add some sort of Conclusion and CTAs (call to action - link to getting started guide, for example).